internal/tiger/cmd/db.go

```diff
@@ -932,3 +960,363 @@ func isConnectionRejected(err error) bool {
 	// should be treated as "unreachable" (exit code 2)
 	return false
 }
+
+// isAuthenticationError checks if the error is a PostgreSQL authentication failure
+func isAuthenticationError(err error) bool {
+	if err == nil {
+		return false
+	}
+	// Check for PostgreSQL error code 28P01 (invalid_password) or 28000 (invalid_authorization_specification)
+	var pgErr *pgconn.PgError
+	if errors.As(err, &pgErr) {
+		return pgErr.Code == "28P01" || pgErr.Code == "28000"
+	}
+	return false
+}
+
+// testConnectionWithPassword tests database connectivity with a specific password
+// Returns nil on success, error on failure
+func testConnectionWithPassword(ctx context.Context, details *common.ConnectionDetails, password string, timeout time.Duration) error {
+	// Build connection string with password (URL-encode the password to handle special chars)
+	connStr := fmt.Sprintf("postgresql://%s:%s@%s:%d/%s?sslmode=require",
+		details.Role, url.QueryEscape(password), details.Host, details.Port, details.Database)
```

Comment:
The `common.ConnectionDetails` struct has a [`.String()` method](https://github.com/timescale/tiger-cli/blob/b936b4a160a1914f50feb9b2ff96179f942a694d/internal/tiger/common/connection.go#L78-L86) that can be used for generating a connection string. Rather than duplicating the connection string building logic here, it might be a little cleaner to just set the `Password` field on `details` and use the `.String()` method to get the connection string.